src/DataStructures/Index.hpp

```diff
@@ -77,6 +79,17 @@ class Index {
   }
   /// \endcond
 
+  // @{
+  /*!
+   * \brief Returns the 1-dimensional index corresponding to the `Dim`
+   * dimensional `index`.
```

Comment:
Add a comment about the choice of fastest-varying direction, perhaps referring to `IndexIterator`. Somewhere there should be a reason documented for this choice.